### PR TITLE
Refactors challenge to be more convenient from consumer perspective

### PR DESCRIPTION
**Pull request by James-Pickett**

### CODE REVIEW

File: pkg/challenge/challenger.go [hunk 1]
```diff
@@ -18,16 +19,105 @@ type OuterChallenge struct {
...
+	return echelper.VerifySignature(counterParty, o.Msg, o.Sig)
+}
+
+func (o *OuterChallenge) RequestData() ([]byte, error) {
+	inner, err := o.inner()
+	if err != nil {
+		return nil, err
+	}
+	return inner.RequestData, nil
+}
```
Comment (directionless) [inline pkg/challenge/challenger.go:32]: I think move the `inner()` call to inside Verify will make callers clean

File: pkg/challenge/responder.go [hunk 2]
```diff
@@ -21,72 +18,50 @@ type OuterResponse struct {
...
+	if err := msgpack.Unmarshal(outerResponseBytes, &outerResponse); err != nil {
+		return nil, err
 	}
+	return &outerResponse, nil
+}
 
-	sealed, pub, err := echelper.SealNaCl(innerResponse, &innerChallenge.PublicEncryptionKey)
-	if err != nil {
-		return nil, fmt.Errorf("sealing challenge inner box: %w", err)
+func UnmarshalOuterResponsePng(outerResponsePngBytes []byte) (*OuterResponse, error) {
```
Comment (directionless) [inline pkg/challenge/responder.go:59]:
```suggestion
func UnmarshalResponsePng(outerResponsePngBytes []byte) (*OuterResponse, error) {
```

File: pkg/challenge/responder.go [hunk 3]
```diff
@@ -21,72 +18,50 @@ type OuterResponse struct {
...
+	PublicSigningKey []byte `msgpack:"publicSigningKey"`
+	ChallengeData    []byte `msgpack:"challengeData"`
+	ResponseData     []byte `msgpack:"responseData"`
+	TimeStamp        int64  `msgpack:"timeStamp"`
+}
 
-	signature, err := echelper.Sign(signer, innerResponse)
-	if err != nil {
-		return nil, fmt.Errorf("signing challenge: %w", err)
+func UnmarshalOuterResponse(outerResponseBytes []byte) (*OuterResponse, error) {
```
Comment (directionless) [inline pkg/challenge/responder.go:51]:
```suggestion
func UnmarshalResponse(outerResponseBytes []byte) (*OuterResponse, error) {
```

File: pkg/challenge/challenger.go [hunk 4]
```diff
@@ -18,16 +19,105 @@ type OuterChallenge struct {
...
+
+	var buf bytes.Buffer
+	if err := krypto.ToPng(&buf, response); err != nil {
+		return nil, fmt.Errorf("encoding data to png: %w", err)
+	}
+
+	return buf.Bytes(), nil
+}
+
+func UnmarshalOuterChallenge(outerChallengeBytes []byte) (*OuterChallenge, error) {
```
Comment (directionless) [inline pkg/challenge/challenger.go:98]:
```suggestion
func UnmarshalChallenge(outerChallengeBytes []byte) (*OuterChallenge, error) {
```

File: pkg/challenge/challenger.go [hunk 5]
```diff
@@ -18,16 +19,105 @@ type OuterChallenge struct {
...
+	var outerChallenge OuterChallenge
+	if err := msgpack.Unmarshal(outerChallengeBytes, &outerChallenge); err != nil {
+		return nil, err
+	}
+	return &outerChallenge, nil
+}
+
+func MarshalOuterChallenge(outerChallenge *OuterChallenge) ([]byte, error) {
+	return msgpack.Marshal(outerChallenge)
+}
```
Comment (directionless) [inline pkg/challenge/challenger.go:108]: Do you think this should me a method on outerChallenge directly?

File: pkg/challenge/challenger.go (hunk 1 above)
Comment (James-Pickett) [reply pkg/challenge/challenger.go]: done

File: pkg/challenge/responder.go (hunk 2 above)
Comment (James-Pickett) [reply pkg/challenge/responder.go]: done

File: pkg/challenge/challenger.go (hunk 4 above)
Comment (James-Pickett) [reply pkg/challenge/challenger.go]: done

File: pkg/challenge/challenger.go (hunk 5 above)
Comment (James-Pickett) [reply pkg/challenge/challenger.go]: done

File: pkg/challenge/challenge.go [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+func (o *OuterChallenge) RequestData() []byte {
+	if o.innerChallenge != nil {
+		return o.innerChallenge.RequestData
+	}
+
+	return nil
+}
+
+func (o *OuterChallenge) TimeStamp() int64 {
```
Comment (directionless) [inline pkg/challenge/challenge.go:45]: I think of timestamp as one word. `Timestamp`

Comment (James-Pickett) [reply pkg/challenge/challenge.go]: updated

**Review by directionless — APPROVED** [review]
Let's GOOOOOOOO

**Pull request merged**